features/entire-workflow-versioning.md

```diff
@@ -0,0 +1,68 @@
+# Entire Workflow Versioning
+
+### Problem statement
+
+The current method of versioning Workflows is complex and error prone.
+By default, there's a good chance that upgrading a worker will result in workflows being stuck due to non-determinism errors.
+Recovering from this situation is not trivial because a user now has histories generated by mulitple versions.
+
+### Proposal
+
+Entire Workflow versioning is an approach that'll greatly simplify the current Workflow versioning situation.
+
+- We add a concept of a central repository for Workflow code
+- Users deploy their Workflows directly to the central repository
```

Comment:
This is very close to serverless/FaaS and I think we should explore the landscape here. Many of them have solved similar versioning issues with uploaded software.

Reply:
I agree, and this proposal takes us a step closer to hosting workflows (if we ever go that direction).
As for bundling, we've pretty much got that figured out in TypeScript, we provide our own bundler (which internally uses webpack).